core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/CreatingBillingProjectMonitor.scala

```diff
@@ -19,155 +22,87 @@ import scala.util.Success
-                  dataAccess.rawlsBillingProjectQuery.insertOperations(operationRecords).map(_ => project)
-                }
-              }
-            } yield {
-              updatedProject
-            }
-
-          case Seq(RawlsBillingProjectOperationRecord(_, gcsDAO.CREATE_PROJECT_OPERATION, _, true, Some(error), _)) =>
-            // create project operation finished with an error
-            logger.debug(s"project ${project.projectName} creation finished with errors: $error")
-            Future.successful(project.copy(status = CreationStatuses.Error, message = Option(error)))
-
-          case operations: Seq[RawlsBillingProjectOperationRecord] if operations.forall(rec => rec.done && rec.errorMessage.isEmpty) =>
-            // all operations completed successfully
-            for {
-              ownerGroupEmail <- UserService.getGoogleProjectOwnerGroupEmail(samDAO, project)
-              computeUserGroupEmail <- UserService.getComputeUserGroupEmail(samDAO, project)
-              updatedProject <- gcsDAO.completeProjectSetup(project, Set(ownerGroupEmail, computeUserGroupEmail)) map {
-                case util.Failure(t) =>
-                  logger.info(s"Failure completing project for $project", t)
-                  project.copy(message = Option(t.getMessage))
-                case Success(_) => project.copy(status = CreationStatuses.Ready)
-              }
-            } yield {
-              updatedProject
-            }
-
-          case operations: Seq[RawlsBillingProjectOperationRecord] if operations.forall(rec => rec.done) =>
-            // all operations completed but some failed
-            val messages = operations.collect {
-              case RawlsBillingProjectOperationRecord(_, operationName, _, true, Some(error), _) => s"Failure enabling api $operationName: ${error}"
-            }
-            val errorMessage = messages.mkString("[", "], [", "]")
-            logger.debug(s"errors enabling apis for project ${project.projectName}: $errorMessage")
-            Future.successful(project.copy(status = CreationStatuses.Error, message = Option(errorMessage)))
-
-          case _ =>
-            // still running
-            Future.successful(project)
-        }
-
-        nextStepFuture.recover {
-          case t: Throwable =>
-            logger.error(s"failure processing new project ${project.projectName.value}", t)
-            project.copy(status = CreationStatuses.Error, message = Option(t.getMessage))
-        }
-      }
-
-
-      _ <- datasource.inTransaction { dataAccess =>
-        // save project updates
-        val updatedProjects = maybeUpdatedProjects.toSet -- projects.toSet
-        dataAccess.rawlsBillingProjectQuery.updateBillingProjects(updatedProjects)
-      }
+  def checkPubSub()(implicit executionContext: ExecutionContext): Future[CreatingBillingProjectMonitorMessage] = {
+    for {
+      psMessages <- pubSubDAO.pullMessages(dmPubSubSubscription, 100)
```

Comment:
I don't care enough to warrant changing anything but since project create is relatively infrequent, it seems like doing them 100 at a time instead of 1 at a time is an over complication. In reality they are likely to come in 1 at a time.

Reply:
yeah i went for a big number assuming that lots of rawlses in fiab would potentially be sharing the same topic, and might have to churn through a lot of messages (esp. in the case of auto-tests)

but if we're putting the hostname in the topic then it doesn't matter so much